torch/_inductor/fx_passes/freezing_patterns.py

```diff
@@ -1,27 +1,74 @@
 import functools
 
 import torch
+from .. import config
 from ..pattern_matcher import (
+    _return_true,
     inference_graph,
     init_once_fakemode,
     PatternMatcherPass,
+    register_graph_pattern,
     register_replacement,
+    stable_topological_sort,
 )
 
 aten = torch.ops.aten
 
+# First pass_patterns[0] are applied, then [1], then [2]
+pass_patterns = [
+    PatternMatcherPass(),
+    PatternMatcherPass(),
+    PatternMatcherPass(),
+]
+
+
+def freezing_passes(gm: torch.fx.GraphModule):
+    """
+    Passes that are applied to the graph to freeze pass.
+    """
+
+    lazy_init()
+    for patterns in pass_patterns:
+        patterns.apply(gm.graph)
+
+    if torch._C._has_mkldnn and config.cpp.weight_prepack:
+        from .mkldnn_fusion import _eliminate_duplicate_packed_nodes
+
+        _eliminate_duplicate_packed_nodes(gm)
+
+    stable_topological_sort(gm.graph)
+    gm.recompile()
+    gm.graph.lint()
+
 
 @init_once_fakemode
-@functools.lru_cache(None)
-def get_freezing_patterns():
-    patterns = PatternMatcherPass()
+def lazy_init():
+    if torch._C._has_mkldnn and config.cpp.weight_prepack:
+        from .mkldnn_fusion import _mkldnn_weight_pack_init
+
+        _mkldnn_weight_pack_init()
 
+    addmm_patterns_init()
+
+
+def register_freezing_graph_pattern(
+    pattern, extra_check=_return_true, pass_number=0, prepend=False
```

Comment:
can you remove the `prepend` call here ? it wasn't actually being used so i removed it from the arg. this will error on rebase (happened to me)

Reply:
Yes, removed.